classes/class-wc-connect-paypal-ec.php

```diff
@@ -67,18 +73,20 @@ public function init() {
 		public function request_body( $body ) {
 			$methods_to_proxy = array( 'SetExpressCheckout', 'GetExpressCheckoutDetails', 'DoExpressCheckoutPayment' );
 			if ( in_array( $body['METHOD'], $methods_to_proxy ) ) {
-				add_filter( 'woocommerce_paypal_express_checkout_request_endpoint', array( $this, 'request_endpoint' ), 10, 2 );
+				add_filter( 'pre_http_request', array( $this, 'proxy_request' ), 10, 3 );
 			} else {
-				remove_filter( 'woocommerce_paypal_express_checkout_request_endpoint', array( $this, 'request_endpoint' ), 10, 2 );
+				remove_filter( 'pre_http_request', array( $this, 'proxy_request' ), 10, 3 );
 			}
 			return $body;
 		}
 
-		/**
-		 * Get WCS PayPal proxy endpoint
-		 */
-		public function request_endpoint( $endpoint, $environment ) {
-			return trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL ) . 'paypal/nvp/' . $environment;
+		public function proxy_request( $preempt, $r, $url ) {
+			if ( ! preg_match( '/paypal.com\/nvp$/', $url ) ) {
+				return false;
```

Comment:
If we aren't going to proxy, we should pass through the value of `$preempt`.

Reply:
Oh, right!